File: Nível 2/1983 - O Escolhido/main.lua
quantia = tonumber(io.read())

matricula = 0
maiorNota = 0

for i = 1, quantia do
    valores = io.read() 
    mat, nota = valores:match("(%S+)%s+(%S+)")
    mat, nota = tonumber(mat), tonumber(nota)

    if nota > maiorNota and nota >= 8 then
        matricula = mat 
        maiorNota = nota 
    end
end

if maiorNota == 0 then
    print("Minimum note not reached")
else
    print(string.format("%.0f", matricula))
end

File: Nível 2/1983 - O Escolhido/main.py
quantia = int(input("A: "))
matricula = 0
maiorNota = 0

for i in range(1, quantia + 1):
    mat, nota = map(float, input("B: ").split(" "))

    if nota > maiorNota and nota >= 8:
        matricula = mat 
        maiorNota = nota

if maiorNota == 0:
    print("Minimum note not reached")
else:
    print(f"{matricula:.0f}")